lib/SILOptimizer/IPO/CapturePromotion.cpp

```diff
@@ -977,7 +977,7 @@ bool isPartialApplyNonEscapingUser(Operand *CurrentOp, PartialApplyInst *PAI,
   unsigned Index = OpNo - 1 + closureConv.getNumSILArguments();
 
   auto *Fn = PAI->getReferencedFunction();
-  if (!Fn || !Fn->isDefinition()) {
+  if (!Fn || !Fn->isDefinition() || Fn->isDynamicallyReplaceable()) {
```

Comment:
Please comment explaining why it this pass in particular cannot safe look at the dynamically replaceable body.